test_launch_ros/test/test_launch_ros/test_track_node_names.py

```diff
@@ -101,7 +101,7 @@ def test_launch_node_with_name_without_namespace():
     assert get_node_name_count(context, f'/{TEST_NODE_NAME}') == 1
 
 
-def test_launch_composable_node_with_names():
+def test_launch_composable_node_with_names(pytestconfig):
```

Comment:
Though I'm ok with this, doesn't using `capsys` just works?

Reply:
is this related with your previous comment?

> EDIT: I am able to capture the output by using capfd, but pytest also needs to be run with the -s option. I'll look into this further.

Reply:
Yeah, it is related to my previous comment. Using `capsys` and `redirect_stdout()` did not work, as we need to capture the output of subprocesses in this case. Just using `capfd` sort of worked, as I mentioned in my previous comment.

Reply:
>  as we need to capture the output of subprocesses in this case.

It isn't the output of a sub-process.
All launch actions are executed in the same process (and some of those actions create subprocess, but we aren't trying to capture their output here).

Reply:
I think that the issue comes from how we run `pytest`, probably some of the flags to change capturing should be used to avoid this.
Anyways, I'm ok with the current code.

Reply:
@wjwwood tests are passing with this as-is, do you think it's fine to merge?

Reply:
Sure.